x/eth-bifrost/IMPLEMENTATION_SPEC.md

```diff
@@ -0,0 +1,133 @@
+# The THORChain <--> Ethereum bifrost
+
+## Introduction
+
+This document acts as a guide to implementors of the THORChain/Ethereum Bifrost to allow for Ethereum and ERC-20 tokens to move between THORChain and Ethereum.
+
+## Related Reading
+This spec is heavily based on the Cosmos Peggy project spec (https://github.com/cosmos/peggy/blob/master/spec/readme.md) and new Cosmos oracle module (https://github.com/cosmos/cosmos-sdk/blob/master/examples/democoin/x/oracle/README.md). Make sure to read and understand both of those first.
+
+Code from those projects will be relevant too, but the license has not yet been confirmed for Peggy (see https://github.com/cosmos/peggy/issues/44). The Solidity Ethereum smart contract code looks fairly usable, but may require some adaptation and additions. The Peggy Go code is mostly boilerplate and not really useful. The Rust witness code should probably not be used. The Cosmos oracle module looks usable, but may require some adaptation and additions.
+
+The spec is written under the assumption that the LICENCE for Peggy will be Apache 2.0, like the Cosmos SDK, or that all code can be easily used. The project has not been completed nor updated in the last few months. I suggest contacting the Peggy team and possibly even considering asking for a grant to do the work of getting it completed.
+
+## Relation to main Bifrost paper
+The main Bifrost paper describes a more complex and possibly more secure system with a custom `m of n` validator signatures required, support for a changing validator set, random reshuffling and VRFs, SPV proofs and gossip-layer communication. The spec is constrained to a very simple MVP that can be used for a Testnet and get real users running and playing with it asap, and does not include any of those features or security/performance improvements.
+
+## Overview
+The Bifrost MVP will add functionality to THORChain that allows the movement of Ether and ERC20 from Ethereum to THORChain and back, where they are represented as ERC20 tokens. It will not support movement of RUNE or THORChain native tokens to Ethereum.
+
+The MVP is split into three components:
+ - the Ethereum smart contract for sending tokens into the Bifrost from Ethereum
+ - the current THORChain Full Node and Validator, with additional bifrost and oracle modules
+ - a signer process, that detects Bifrost requests to send to Ethereum, and signs them appropriately for Ethereum
```

Comment:
Why do we need the signer process? A tx that is in the chain has already be signed by >2/3 of our validators. Isn't the tx with the signatures (maybe a smaller merkle-proof package) enough proof for the smart contract to unlock/mint?

Reply:
Got it, thanks!